xtask/src/check.rs

```diff
@@ -1,5 +1,80 @@
 use crate::prelude::*;
 
+const EXTRA_LINTS: &[&str] = &[
+    // == Safer unsafe == //
+    "unsafe_op_in_unsafe_fn",
+    "invalid_reference_casting",
+    "pointer_structural_match",
+    "clippy::undocumented_unsafe_blocks",
+    "clippy::multiple_unsafe_ops_per_block",
+    "clippy::transmute_ptr_to_ptr",
+    "clippy::as_ptr_cast_mut",
+    "clippy::cast_ptr_alignment",
+    "clippy::fn_to_numeric_cast_any",
+    "clippy::ptr_cast_constness",
+    // == Correctness == //
+    "unused_tuple_struct_fields",
+    "clippy::arithmetic_side_effects",
+    "clippy::cast_lossless",
+    "clippy::cast_possible_truncation",
+    "clippy::cast_possible_wrap",
+    "clippy::cast_sign_loss",
+    "clippy::float_cmp",
+    "clippy::as_underscore",
+    // TODO: "clippy::unwrap_used", // let’s either handle `None`, `Err` or use `expect` to give a reason
+    "clippy::large_stack_frames",
+    // == Style, readability == //
+    "absolute_paths_not_starting_with_crate",
+    "single_use_lifetimes",
+    "unreachable_pub",
```

Comment:
Protects us from leaving unreachable `pub` items. We should use `pub(crate)` or no `pub` at all for implementation details. In some cases it’s actually desirable to make a type `pub` without it being reachable (effectively preventing the user from naming this type), but it’s rare enough that we can just explicitly add an `#[allow]` annotation when necessary.